File: 2022/09/18870.js
const numbers = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")[1].split(" ").map(Number);
const sortedNumbers = [...new Set(numbers)].sort((a, b) => a - b);
const hashTable = new Map();
sortedNumbers.forEach((e, i) => hashTable.set(e, i));
const answer = numbers.map((e) => hashTable.get(e));
console.log(answer.join(" "));

File: 2022/09/18870.py
input = __import__('sys').stdin.readline
N = input()
numbers = list(map(int, input().split()))

SortedNumbers = sorted(set(numbers))
hashtable = {e: i for i, e in enumerate(SortedNumbers)}

print(' '.join(map(lambda x: str(hashtable[x]), numbers)))
